File: VB/CustomParameters/MyParameter.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports DevExpress.XtraReports.Parameters
Imports System.ComponentModel

Namespace CustomParameters
	Public Class MyParameter
		Inherits Parameter
		Private _valueMember As String
		Private _displayMember As String
		Private _statement As String

		<DisplayName("Value Member"), Browsable(True), DefaultValue(""), Category("Data")> _
		Public Property ValueMember() As String
			Get
				Return _valueMember
			End Get
			Set(ByVal value As String)
				_valueMember = value
			End Set
		End Property
		<DisplayName("Display Member"), Browsable(True), DefaultValue(""), Category("Data")> _
		Public Property DisplayMember() As String
			Get
				Return _displayMember
			End Get
			Set(ByVal value As String)
				_displayMember = value
			End Set
		End Property
		<DisplayName("Statement"), Browsable(True), DefaultValue(""), Category("Data")> _
		Public Property Statement() As String
			Get
				Return _statement
			End Get
			Set(ByVal value As String)
				_statement = value
			End Set
		End Property
	End Class
End Namespace

File: CS/CustomParameters/MyParameter.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.XtraReports.Parameters;
using System.ComponentModel;

namespace CustomParameters
{
    public class MyParameter : Parameter
    {
        string _valueMember;
        string _displayMember;
        string _statement;

        [DisplayName("Value Member"), Browsable(true), DefaultValue(""), Category("Data")]
        public string ValueMember
        {
            get { return _valueMember; }
            set { _valueMember = value; }
        }
        [DisplayName("Display Member"), Browsable(true), DefaultValue(""),Category("Data")]
        public string DisplayMember
        {
            get { return _displayMember; }
            set { _displayMember = value; }
        }
        [DisplayName("Statement"), Browsable(true), DefaultValue(""), Category("Data")]
        public string Statement
        {
            get { return _statement; }
            set { _statement = value; }
        }
    }
}
